src/notebooks/controllers/controllerDefaultService.ts

```diff
@@ -17,6 +17,9 @@ import {
     IVSCodeNotebookController
 } from './types';
 
+/**
```

Comment:
The `ControllerPreferredService` "Computes and tracks the preferred kernel for a notebook", what's the major difference between "default" and "preferred" here?

Reply:
Preferred reads the data in the notebook. Default doesn't.